server/src/main/java/org/opensearch/tasks/TaskManager.java

```diff
@@ -150,6 +153,20 @@ public Task register(String type, String action, TaskAwareRequest request) {
             logger.trace("register {} [{}] [{}] [{}]", task.getId(), type, action, task.getDescription());
         }
 
+        // just register read operations
+        if (action.startsWith("indices:data/read")) {
```

Comment:
I would suggest to enrich action with something like `isResourceTrackingEnabled()` and use it as an indicator of the need to track resources. Also, the tracking key (in this case, `indices` has to be provided by the action as as well, fe as `getResourceTrackingKey` method.